crates/polars-plan/src/dsl/mod.rs

```diff
@@ -966,7 +966,7 @@ impl Expr {
         // we take the not branch so that self is truthy value of `when -> then -> otherwise`
         // and that ensure we keep the name of `self`
 
-        when(self.clone().is_not_nan())
+        when(self.clone().is_not_nan().or(self.clone().is_null()))
```

Comment:
I think we can save a `negate + allocation` if we say `& ca.is_not_null()`. In that case we can use the validity mask directly and convert that to boolean array.